query/src/main/scala/filodb/query/exec/ScalarTimeBasedExec.scala

```diff
@@ -0,0 +1,80 @@
+package filodb.query.exec
+
+import scala.concurrent.duration.FiniteDuration
+
+import monix.eval.Task
+import monix.execution.Scheduler
+import monix.reactive.Observable
+
+import filodb.core.DatasetRef
+import filodb.core.metadata.Column.ColumnType
+import filodb.core.query._
+import filodb.core.store.ChunkSource
+import filodb.query.{BadQueryException, QueryConfig, QueryResponse, QueryResult, ScalarFunctionId}
+import filodb.query.Query.qLogger
+import filodb.query.ScalarFunctionId.{DayOfMonth, DayOfWeek, DaysInMonth, Hour, Minute, Time, Year}
+
+/**
+  * Exec Plans for time functions which can execute locally without being dispatched
+  */
+case class ScalarTimeBasedExec(id: String,
+                               dataset: DatasetRef, params: RangeParams,
+                               function: ScalarFunctionId,
+                               limit: Int,
+                               submitTime: Long = System.currentTimeMillis()) extends LeafExecPlan {
+
+
+  /**
+    * Sub classes should override this method to provide a concrete
+    * implementation of the operation represented by this exec plan
+    * node
+    */
+  override def doExecute(source: ChunkSource, queryConfig: QueryConfig)
+                                  (implicit sched: Scheduler, timeout: FiniteDuration): ExecResult = ???
+
+  /**
+    * Args to use for the ExecPlan for printTree purposes only.
+    * DO NOT change to a val. Increases heap usage.
+    */
+  override protected def args: String = s"params = $params, function = $function"
+
+  override def execute(source: ChunkSource,
+                       queryConfig: QueryConfig)
+                      (implicit sched: Scheduler,
+                       timeout: FiniteDuration): Task[QueryResponse] = {
+    val columns: Seq[ColumnInfo] = Seq(ColumnInfo("timestamp", ColumnType.LongColumn),
```

Comment:
We should probably just have a fixed Object constant for this schema since it's used so often.  Instead of allocating a new one every time.